### PR TITLE
Use clean PF components for building Overview cards

### PR DESCRIPTION
**Pull request by rawagner**
screencapture-localhost-9000-dashboards-2021-11-09-09_50_37

This PR also exposes Overview and OverviewGrid components via dynamic plugin API & removes Dashboard/DashboardGrid from internal one

### CODE REVIEW

**Comment by rawagner** [comment]
@andybraren @mceledonia can you take a look ?

**Comment by spadgett** [comment]
@bipuladh FYI

**Comment by spadgett** [comment]
Still working through the code, but the changes look good overall 👍 

/assign @yapei 
for QE review

@yapei we need a regression test of the home / project / node overviews

**Comment by bipuladh** [comment]
A bunch of components will now be lost from the SDK, which were being consumed in 4.9(compatibility build for 4.10). References for these components need to be updated. cc @shdn-ibm

**Comment by yanpzhan** [comment]
@rawagner Do you meet errors when build code against the pr code? I met 5 errors when build code in console repo: http://pastebin.test.redhat.com/1007523

**Comment by rawagner** [comment]
> @rawagner Do you meet errors when build code against the pr code? I met 5 errors when build code in console repo: http://pastebin.test.redhat.com/1007523

@yanpzhan I see those erros, but they are present on master too. Not related to changes in this PR.
cc @glekner

**Comment by glekner** [comment]
@yanpzhan @rawagner https://github.com/openshift/console/pull/10432 will fix. could take time though

**Comment by andybraren** [comment]
Ideally the amount of space to the right of the "View X" links in the top-right corner of most cards would be the same as the amount of space to the left of each card's name. As Sam mentioned it looks like those links are wrapped in Button components - maybe unwrapping them would fix that spacing?

Same for the "1 hour" dropdown/select component in the Utilization card, if it's possible natively. If not no biggie for this PR.

**Comment by rawagner** [comment]
@spadgett @andybraren 
removing `Button` fixes the spacing issues, but breaks alignment with the title
localhost_9000_dashboards (1)

Im not sure if there is a way how to fix all problems without tweaking css.

**Comment by rawagner** [comment]
Seems like removing `self-align` prop from PF's `CardActions` fixes the issue. Reported at https://github.com/patternfly/patternfly-react/issues/6673

**Comment by rawagner** [comment]
As a workaround I've added a css class which sets `align-self: center` . It will be removed once PF issue is resolved.

@spadgett @andybraren can you take a look again ?

File: frontend/public/style/_common.scss [hunk 1]
```diff
@@ -566,3 +566,21 @@ dl.co-inline {
...
+    left: 0;
+    min-height: 2rem;
+    pointer-events: none;
+    position: absolute;
+    width: calc(100% - var(--pf-global--spacer--lg));
+  }
+}
+
+.co-overview-card__actions {
+  align-self: center !important;
```
Comment (spadgett) [inline frontend/public/style/_common.scss:585]: Let's open an issue or bug to track when we can remove this.

**Review by spadgett — APPROVED** [review]
/lgtm

File: frontend/public/style/_common.scss (hunk 1 above)
Comment (andybraren) [reply frontend/public/style/_common.scss:585]: Agreed, thanks Rasto!

**Comment by rawagner** [comment]
/hold
e2e tests are failing

**Comment by rawagner** [comment]
/label docs-approved
/label px-approved

Nothing to document here, it just a visual update to align with PF

**Review by jtomasek — APPROVED** [review]
/lgtm

**Pull request merged**